agent/src/agent/pipeline/config_handlers/base.py

```diff
@@ -60,11 +60,26 @@ def override_base_config(self, client_config, new_uuid=None, new_pipeline_title=
         return self.config
 
     def update_source_configs(self):
-        if 'library' in self.client_config['source']['config']:
-            self.config['stages'][0]['library'] = self.client_config['source']['config']['library']
+        if 'library' in self.pipeline.source.config:
+            self.config['stages'][0]['library'] = self.pipeline.source.config['library']
+        self.__override_source_using_pipeline()
+        self.__write_source_config_to_stage()
+
+    def __override_source_using_pipeline(self):
+        for k, v in self.pipeline.override_source.items():
+            if not self.__is_in_base_source_config(k):
```

Comment:
Why do you need to check it if the same thing is going to be checked for all config on line 81?